core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -762,16 +765,25 @@ private String fullTableName(TableIdentifier ident) {
     return String.format("%s.%s", name(), ident);
   }
 
+  private FileIO newFileIO(Map<String, String> properties) {
+    if (null != ioBuilder) {
+      return ioBuilder.apply(properties);
+    } else {
+      String ioImpl = properties.get(CatalogProperties.FILE_IO_IMPL);
+      return
+          CatalogUtil.loadFileIO(
+              ioImpl != null ? ioImpl : ResolvingFileIO.class.getName(), properties, conf);
```

Comment:
nit: maybe worth inlining this, so that we don't need the null check on `ioImpl`:
```
CatalogUtil.loadFileIO(
          properties.getOrDefault(CatalogProperties.FILE_IO_IMPL, ResolvingFileIO.class.getName()),
          properties,
          conf);
```

Reply:
I updated this to be more like the JDBC catalog, where we use `getOrDefault` on the line above to clean it up.